File: src/main/scala/RegistryServices.scala
package codecraft.registry

import codecraft.codegen.CmdGroupConsumer

trait IRegistryStore extends CmdGroupConsumer {
  def delete(cmd: DeleteRegistry): DeleteRegistryReply
  def get(cmd: GetRegistries): GetRegistriesReply
  def put(cmd: PostRegistry): PostRegistryReply
  val methodRegistry = Map[String, Any => Any](
    "cmd.registry.delete" -> {
      any => delete(any.asInstanceOf[DeleteRegistry])
    },
    "cmd.registry.get" -> {
      any => get(any.asInstanceOf[GetRegistries])
    },
    "cmd.registry.put" -> {
      any => put(any.asInstanceOf[PostRegistry])
    }
  )
}

File: src/main/fsharp/RegistryServices.fs
namespace codecraft.registry

open codecraft.codegen

[<AbstractClass>]
type IRegistryStore () =
  inherit CmdGroupConsumer()
  override this.methodRegistry =   Map<string, obj -> obj> [
      ("cmd.registry.delete", fun (any: obj) -> this.delete(any :?> DeleteRegistry) :> obj)
      ("cmd.registry.get", fun (any: obj) -> this.get(any :?> GetRegistries) :> obj)
      ("cmd.registry.put", fun (any: obj) -> this.put(any :?> PostRegistry) :> obj)
    ]
  abstract member delete: DeleteRegistry -> DeleteRegistryReply
  abstract member get: GetRegistries -> GetRegistriesReply
  abstract member put: PostRegistry -> PostRegistryReply
